# Abhinav Adduri
[\[LinkedIn\]](https://www.linkedin.com/in/abhinadduri/) &nbsp; [\[CV\]](https://raw.githubusercontent.com/abhinadduri/abhinadduri.github.io/main/assets/Abhinav_CV.pdf) &nbsp; [\[aadduri@andrew.cmu.edu\]](mailto:aadduri@andrew.cmu.edu) 
<img src="assets/images/abhinav-portrait.jpg" align="right" alt="Abhinav Adduri"/>

Hello! I am a 5th year Ph.D. Candidate at CMU advised by [Dr. Hosein Mohimani](http://mohimanilab.cbd.cmu.edu/). My research focus is on applying high performance computing techniques and AI for computational drug discovery.

Organisms in nature such as bacteria or fungi can produce drugs that can benefit humanity, but as of now, finding these drugs is laborious and mostly trial and error. Since DNA largely dictates the characteristics of organisms, my research started with the question: can we examine the DNA of microbes to reliably predict the drug-like compounds they will create? And, can we characterize whether they will be useful to humans as antibiotics, antifungals, or even as cancer therapeutics? 

Our lab is using the latest in LLMs and bioinformatics to answer these questions, starting from DNA and ending with wet lab validation of our results. We are working with *Nature* to publish our most recent discovery: an antifungal that we discovered which kills a drug-resistant 
fungal strain that the CDC considers an urgent threat. To address the need for high performance in this field, we make heavy use of [Rust](https://www.rust-lang.org), which in tandem with efficient algorithm design has led to order of magnitude improvements in runtimes.

I am also working on using self-supervised techniques on multi-modal biological data to tackle problems downstream of drug discovery, such as drug-target interaction and matching small molecules to their mass spectra.

Before starting my Ph.D., I majored in Computer Science at UC Berkeley.

# Publications
Yan, D.\*, Zhou, M.\*, **Adduri, A.\***, Guler, M., Liu, S., Shin, H., Kovach, T., Oh, G., Liu, X., Deng, Y., Cao, L., Behsaz, B., & Mohimani, H. (Expected 2024). Seq2PKS: Discovering modular type I cis-AT polyketide natural products by integrating computational mass spectrometry and genome mining. *Manuscript under review for publication in Nature Communications.*

**Adduri, A.**, & Kim, S. (2023). Ornaments for accurate and efficient allele-specific expression estimation with bias correction. *Manuscript under review for publication in The American Journal of Human Genetics. [[ArXiv](https://www.biorxiv.org/content/10.1101/2023.10.25.564046v1.full)]*

Mongia, M., Baral, R., **Adduri, A.**, Yan, D., Liu, Y., Bian, Y., Kim, P., Behsaz, B., & Mohimani, H. (2023). AdenPredictor: Accurate prediction of the adenylation domain specificity of nonribosomal peptide biosynthetic gene clusters in microbial genomes. *Bioinformatics, 39*(Supplement 1), i40-i46.

**Adduri, A.** (2019). IoT integration in manufacturing processes. In *Simulation for Industry 4.0: Past, Present, and Future* (pp. 129-139).

**Adduri, A.**, & Schruben, L. (2017). The Tao of simulation. In *2017 Winter Simulation Conference (WSC)* (pp. 607-616). IEEE.

*Equal Contribution.

# Pending publications... stay tuned!

Behsaz, B.\*, **Adduri, A.\***, Guler, M.\*, Mohamed, O. G., Caraballo-Rodríguez, A. M., Duggan, B., Chaudhary, N., Miller, C., Liu, S., Krummenacher, B., Broders, K., Zamith-Miranda, D., Schultz, P. J., Sherman, D. H., Clement, J., Nakayasu, E., Nosanchuk, J., Dorrestein, P. C., Tripathi, A., & Mohimani, H. (Expected 2024). Pathogen-oriented platform for large-scale natural product discovery identifies novel antifungal targeting drug-resistant candidiasis. *Manuscript in preparation for submission to Nature.*

**Adduri, A.**, Yan, D., Guler, M., Suraparaju, C., Fang, N., Tripathi, A., & Mohimani, H. (Expected 2024). Seq2Hybrid: Discovering type I polyketide and non-ribosomal peptide hybrid natural products. *Manuscript in preparation for submission to Nature Communications.*

**Adduri, A.\***, McNutt, A.\*, Suraparaju, C., Fang, N., Koes, D., Mohimani, H. (Expected 2024). Protein foundation models for accurate and generalizable adenylation domain specificity prediction. *Manuscript in preparation for submission to NeurIPS.*

*Equal Contribution.

# Prior Work / Internships

|     Company      |       Location       |           Dates           |
|------------------|----------------------|---------------------------|
| Square / Caviar  | San Francisco, CA    | May 2019 - August 2019    |
| Rubrik           | Palo Alto, CA        | May 2018 - August 2018    |
| Mozilla          | Portland, OR         | May 2017 - August 2017    |
| Bio-G            | Berkeley, CA         | May 2016 - April 2017     |

# Patents

[Automated Review of Source Code for Style Issues](https://patents.justia.com/patent/11294792), Rubrik 2021

# Side Projects
[RecursiveGPT](https://github.com/james-julius/recursive-gpt) - a chat interface I helped develop at an AGI house hackathon which searches for adjacent GPT conversations in embedding space and suggests follow-up questions to enhance your knowledge dives.

[kilter.net](https://github.com/mdayao/kilter.net) - a convolutional variational autoencoder to generate climbing routes conditioned on difficulty and board angle.

[CMU Coffee Chats](https://github.com/scs-phd-deans-committee/coffee-chats-website) - a simple website to connect like-minded CMU students for conversations over coffee ☕.

[Firefox Send](https://github.com/mozilla/send) - a peer-to-peer, private file sharing app I developed during my time at Mozilla which was covered by Forbes, TechRepublic, etc.

[Tao](https://github.com/abhinadduri/tao) - a browser based discrete event simulation engine with IoT capabilities.

